superset/db_engine_specs/base.py

```diff
@@ -1024,7 +1024,7 @@ def estimate_query_cost(
         if not cls.get_allow_cost_estimate(extra):
             raise Exception("Database does not support cost estimation")
 
-        user_name = g.user.username if g.user else None
+        user_name = g.user.username if g.user and hasattr(g.user, "username") else None
```

Comment:
it probably makes more sense to just check if the user is authenticated, using `g.user.is_authenticated`

Reply:
I originally was checking `g.user.is_anonymous` in these use cases, but opted for the explicit check for username, as this is the value that is being fetched. There _may_ be use cases where username is not present in other states down the road, so thought explicit `hasattr` would be more future-proof.

Reply:
ok